hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/sink/TestWriteMergeOnRead.java

```diff
@@ -41,6 +41,8 @@ public class TestWriteMergeOnRead extends TestWriteCopyOnWrite {
   @Override
   protected void setUp(Configuration conf) {
     conf.setBoolean(FlinkOptions.COMPACTION_ASYNC_ENABLED, false);
+    // test legacy stream write function
+    conf.setBoolean(FlinkOptions.INSERT_ROWDATA_MODE_ENABLED, false);
```

Comment:
Can we fix the test operator inference instead of the flag.

Reply:
> fix the test operator inference

Do you mean TestData#getWritePipeline? Here adding the config option is intended to make the test operator inference choose legacy write function.

Reply:
remove this change and also the class `TestRowDataWriteMergeOnRead`.